### ISSUE TITLE
Adversarial Patch Pytorch Yolo Attack Training impacts the Detection Model

### ISSUE THREAD

**Issue by xHascox** [Trusted-AI/adversarial-robustness-toolbox#2601]
**Describe the bug**
When using `notebooks/adversarial_patch/attack_adversarial_patch_pytorch_yolo.ipynb`, training a patch also impacts the object detection model resulting in lowered accuracy even on benign images (without a patch).

**To Reproduce**
1. Open the Notebook on Colab: [https://colab.research.google.com/drive/1AD0by6dkiMYUwHz18gDsAEPArbT6sltJ?usp=sharing](https://colab.research.google.com/drive/1AD0by6dkiMYUwHz18gDsAEPArbT6sltJ?usp=sharing) or download [attack_adversarial_patch_pytorch_yolo_issue.json](https://github.com/user-attachments/files/19738553/attack_adversarial_patch_pytorch_yolo_issue.json) and rename to .ipynb due to filetype limitation
2. Optional: Click on Runtime -> Run All
3. Scroll down to Section `Detector Predictions on Benign Images` and compare these to the following:
4. Scroll down to Section `Predictions on patched images` and look at the images with title "Detector after Patch Training"
5. Notice the differences in predictions, e.g: The left cat is a dog, a 'person' box left to the person in the kitchen, and a lamp is detected as a frisbee. (as shown in the Screenshots below)
6. In Section `Detector freshly initialized` the detector is loaded newly and we get the expected output again.

Note: It seems that at least about 100 training steps are necessary for noticeable results.
Note: The provided Notebook is a slight adaptation of the notebook provided in the ART repo: `notebooks/adversarial_patch/attack_adversarial_patch_pytorch_yolo.ipynb`

**Expected behavior**
Training a patch should only impact the patch. The object detection model that is attacked should not change during patch training.

**Screenshots**
Before Training:
![Image](https://github.com/user-attachments/assets/119faf78-29e8-48b8-94a6-0ffab7e4525c)

After Training:
![Image](https://github.com/user-attachments/assets/4f964905-6990-4f91-8ed0-37eb16e9daa0)

**System information:**
 - OS: Linux (Google Colab)
 - Python version: 3.11.12
 - ART version: 1.19.1
 - Pytorch: 2.6.0+cu124

**Comment by beat-buesser** [comment]
Hi @xHascox Thank you very much for discovering and reporting this bug! I'll include the fix provided in #2663 in ART 1.19.2.
